### ISSUE TITLE
AttributeError: 'Llama' object has no attribute 'ctx'

### ISSUE THREAD

**Issue by hippalectryon-0** [su77ungr/CASALIOY#33]
Using the default conf, I run `python .\ingest.py` and get

```
Traceback (most recent call last):
  File "C:\Users\Hippa\PycharmProjects\CASALIOY\ingest.py", line 51, in <module>
    main(sources_directory, cleandb)
  File "C:\Users\Hippa\PycharmProjects\CASALIOY\ingest.py", line 43, in main
    llama = LlamaCppEmbeddings(model_path=llama_embeddings_model, n_ctx=model_n_ctx)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "pydantic\main.py", line 339, in pydantic.main.BaseModel.__init__
  File "pydantic\main.py", line 1102, in pydantic.main.validate_model
  File "C:\Users\Hippa\PycharmProjects\CASALIOY\venv\Lib\site-packages\langchain\embeddings\llamacpp.py", line 98, in validate_environment
    raise NameError(f"Could not load Llama model from path: {model_path}")
NameError: Could not load Llama model from path: models/ggml-model-q4_0_new.bin
Exception ignored in: <function Llama.__del__ at 0x000001F97F879E40>
Traceback (most recent call last):
  File "C:\Users\Hippa\PycharmProjects\CASALIOY\venv\Lib\site-packages\llama_cpp\llama.py", line 1060, in __del__
    if self.ctx is not None:
       ^^^^^^^^
AttributeError: 'Llama' object has no attribute 'ctx'
```

**Comment by hippalectryon-0** [comment]
Closing - I'm tired, I forgot I renamed the models

**Comment by hippalectryon-0** [comment]
It seems like it can't load the model, but it's clearly there. I'm investigating.

**Comment by su77ungr** [comment]
Try absolute paths. Did you intend to use the **new.bin and changed the .env. ?

**Comment by hippalectryon-0** [comment]
I got that error on two computers, but for different reasons.

First time I got it because of a typo on the path. That's the error message in my first comment here.

But then I got more or less the same error ! It works with my ggjt embeddings, but not with the default ggml embeddings. I'm still investigating.

**Comment by su77ungr** [comment]
We might want to consider using only one model run to simplify this and add redundancy later. Maybe we'll find a port for Mosaic convertion too.

**Comment by hippalectryon-0** [comment]
Update: This "new" bug comes from updating • Updating langchain (0.0.166 -> 0.0.168)
  • Updating llama-cpp-python (0.1.49 -> 0.1.50)

**Comment by su77ungr** [comment]
@hippalectryon-0 Also found this [MosaiML GGML](https://huggingface.co/LLukas22/mpt-7b-ggml)

Give it a try with them.

**Comment by hippalectryon-0** [comment]
Should be fixed by setting `llama-cpp-python = "<=0.1.49"`. Will be in my next PR.

For the MosaiML: haven't tried yet, feel free to create another issue so that we don't forget after closing this one
